shared_model/validators/field_validator.hpp

```diff
@@ -48,27 +48,37 @@ namespace shared_model {
           ReasonsGroupType &reason,
           const interface::types::AccountIdType &account_id) const {
         if (not std::regex_match(account_id, account_id_)) {
-          reason.second.push_back("Wrongly formed account_id");
+          auto message =
+              generateErrorMessage("Wrongly formed account_id",
+                                   static_cast<std::string>(account_id));
+          reason.second.push_back(std::move(message));
         }
       }
 
       void validateAssetId(
           ReasonsGroupType &reason,
           const interface::types::AssetIdType &asset_id) const {
         if (not std::regex_match(asset_id, asset_id_)) {
-          reason.second.push_back("Wrongly formed asset_id");
+          auto message = generateErrorMessage(
+              "Wrongly formed asset_id", static_cast<std::string>(asset_id));
+          reason.second.push_back(std::move(message));
         }
       }
 
       void validateAmount(ReasonsGroupType &reason,
                           const interface::Amount &amount) const {
-        /* put here any validations*/
+        if (amount.intValue() == 0) {
+          reason.second.push_back("Amount must be greater than 0");
+        }
       }
 
       void validatePubkey(ReasonsGroupType &reason,
                           const interface::types::PubkeyType &pubkey) const {
-        if (pubkey.blob().size() != 32) {
-          reason.second.push_back("Public key has wrong size");
+        if (pubkey.blob().size() != key_size) {
+          auto message =
+              generateErrorMessage("Public key has wrong size",
```

Comment:
*couldhave*
IMO `generateErrorMessage` can be just alias for `boost::format`. It is not very good to use `std::to_string`. What if I, as a developer of validator, want to make some comprehensive error? Like, `str+number+str+hex+number` (I hope you get the idea)

Reply:
Why is std::to_string bad?